src/ir/item.rs

```diff
@@ -681,6 +681,34 @@ impl Item {
         }
     }
 
+    /// Create a fully disambiguated name for an item, including template
+    /// parameters if it is a type
+    pub fn full_disambiguated_name(&self, ctx: &BindgenContext) -> String {
+        let mut s = "".into();
+        let level = 0;
+        self.push_disambiguated_name(ctx, &mut s, level);
+        s
+    }
+
+    /// Helper function for full_disambiguated_name
+    fn push_disambiguated_name(&self, ctx: &BindgenContext, to: &mut String, level: u8) {
+        to.push_str(&self.canonical_name(ctx));
+        match *self.kind() {
+            ItemKind::Type(ref ty) => match *ty.kind() {
+                TypeKind::TemplateInstantiation(ref inst) => {
+                    to.push_str(&format!("_open{}_", level));
+                    for arg in inst.template_arguments() {
+                        ctx.resolve_item(*arg).push_disambiguated_name(ctx, to, level + 1);
```

Comment:
This won't work for nested templates, afaict, because you don't handle `ResolvedTypeRef`, or anything like that.